src/compiler/transformers/add-static-style.ts

```diff
@@ -134,16 +134,67 @@ const createStyleLiteral = (cmp: d.ComponentCompilerMeta, style: d.StyleCompiler
   return ts.factory.createStringLiteral(style.styleStr);
 };
 
-const createStyleIdentifierFromUrl = (cmp: d.ComponentCompilerMeta, style: d.StyleCompiler) => {
-  style.styleIdentifier = dashToPascalCase(cmp.tagName);
-  style.styleIdentifier = style.styleIdentifier.charAt(0).toLowerCase() + style.styleIdentifier.substring(1);
-
+export const createStyleIdentifier = (cmp: d.ComponentCompilerMeta, style: d.StyleCompiler) => {
+  /**
+   * remove duplicates in `styleUrl`
+   */
+  const externalStyles = Array.from(new Set(style.externalStyles.map((s) => s.absolutePath)));
+  /**
+   * Set a styleIdentifier which will be propagated to to the component and
+   * later picked up by rollup when it injects the parsed CSS directly into
+   * the component, see `compilerCtx.worker.transformCssToEsm` in
+   * `src/compiler/bundle/ext-transforms-plugin.ts`
+   */
+  style.styleIdentifier = dashToPascalCase(cmp.tagName.charAt(0).toLowerCase() + cmp.tagName.substring(1));
   if (style.modeName !== DEFAULT_STYLE_MODE) {
     style.styleIdentifier += dashToPascalCase(style.modeName);
   }
+  return createStyleIdentifierFromUrl(style.styleId, externalStyles);
+};
 
-  style.styleIdentifier += 'Style';
```

Comment:
For my own understanding, why do we remove this? I vaguely recall it helps disambiguate from other identifiers, but maybe I'm making that up? 🤔

Reply:
Oh, I missed this when partially reverting the change. Added it back. I removed it originally as I thought that creating the identifier based on the file path should be unique enough.